lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -601,17 +605,16 @@ def _group_name(path) -> str:
         Args:
             path: path to rule file.
         """
-        relpath = os.path.relpath(os.path.dirname(path), dir_path)
+        relpath = os.path.relpath(os.path.dirname(path), str(dir_path))
 
         # Generate group name:
         #  - prefix, from the relative path of the rule file;
         #  - name, from juju topology
-        return (
-            f"{'' if relpath == '.' else relpath.replace(os.path.sep, '_') + '_'}"
-            f"{topology.identifier}_alerts"
+        return "{}{}_alerts".format(
+            "" if relpath == "." else relpath.replace(os.path.sep, "_") + "_", topology.identifier
```

Comment:
While you're at it, @mmanciop correctly pointed out that it makes more sense to have the topology identifier first, and the rest to follow.